### PR TITLE
Added support for Proxy Servers

### LINKED ISSUE #4: Allow for configuring HTTP Proxy Servers
If the customer needs to connect to the internet via a Proxy Server we would have to configure a file in
/etc/systemd/system/rhcd.service.d

proxy.conf

Contents
[Service]
Environment="HTTP_PROXY=xxx.xxx.xxxx.xxx:nnn"
Environment="HTTPS_PROXY=xxx.xxx.xxxx.xxx:nnn"

The same values should be set in the catalog.toml file for the 2 proxy values
https://github.com/ansible/automation_services_catalog_client/blob/devel/roles/automation_services_catalog_client/templates/catalog.toml.j2

We need a new section
[Proxies]
HTTP_PROXY="xxx.xxx.xxx.xxx:nnn"
HTTPS_PROXY="xxxx.xxx.xxx:nnn"

### PR DESCRIPTION
**Pull request by mkanoor**
http_proxy
https_proxy
no_proxy

Set in the catalog.toml and /etc/systemd/system/rhcd.service.d/proxy.conf

Fixes #4

### CODE REVIEW

**Comment by mkanoor** [comment]
@Spredzy Please review

**Review by Spredzy — COMMENTED** [review]
@mkanoor looks good to me. Do you mind renaming the variables to a different name ?

* `http_proxy` -> `servicescatalog_http_proxy`
* `https_proxy` -> `servicescatalog_https_proxy`
* `no_proxy` -> `servicescatalog_no_proxy`

I will do it also for other variables in a subsequent PR, but if you could update your PR with those new names.

Then we are good to merge for me.

**Comment by mkanoor** [comment]
@Spredzy I have updated it

**Review by Spredzy — APPROVED** [review]
LGTM

File: roles/automation_services_catalog_client/templates/catalog.toml.j2 [hunk 1]
```diff
@@ -12,3 +12,8 @@ client_key="{{ system_key_file }}"
 [logger]
 # valid levels are error, info, debug, warn
 level="{{ log_level }}"
+
+[PROXIES]
+HTTP_PROXY="{{ servicescatalog_http_proxy }}"
```
Comment (Spredzy) [inline roles/automation_services_catalog_client/templates/catalog.toml.j2:17]: Actually one question: what happens if we do not specify any http_proxy settings?

Would

```
[PROXIES]
HTTP_PROXY=
```

be valid ? accurate ?

File: roles/automation_services_catalog_client/templates/proxy.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[Service]
```
Comment (Spredzy) [inline roles/automation_services_catalog_client/templates/proxy.conf.j2:1]: Same here, say we don't provid them, file ends up looking like

```
[Service]
Environment="HTTPP_PROXY="
```

Is that valid/accurate ?

**Review by Spredzy — COMMENTED** [review]
Few questions with regard to how in case they are not specified the related section should be handled.

I wonder if this should be conditional like

```
{% if sservicescatalog_http_proxu %}
<write my line>
{% endif %}
```

Let me know your thourhgs @mkanoor

File: roles/automation_services_catalog_client/templates/catalog.toml.j2 (hunk 1 above)
Comment (mkanoor) [reply roles/automation_services_catalog_client/templates/catalog.toml.j2:17]: @Spredzy if its blank it works it gets ignored

**Pull request merged**